.github/workflows/test.yml

```diff
@@ -1,6 +1,11 @@
 name: CI
 
-on: [push, pull_request, workflow_dispatch]
+on: 
```

Comment:
did you mean to change this to only run on master? it seems like this should run on any push.

nit: whitespace at EOL

Reply:
This is a quirk of GitHub actions. I previously had just `push` but that does not work for pushes to forks, so I added `pull_request` but now if I (and only people that can push directly to this repo) submit a pull request the tests get run twice: Once for `push` and again for `pull_request`.
There should be no other branches other than master that should need to run the test unless you are creating a pull request. If you really need to then that's what `workflow_dispatch` will allow, but is a manual process.